build-system/tasks/presubmit-checks.js

```diff
@@ -45,11 +43,65 @@ var forbiddenTerms = {
   'indexedDB': '',
   'openDatabase': '',
   'requestFileSystem': '',
-  '\\.scrollTop': '',
   'webkitRequestFileSystem': '',
   'debugger': '',
 };
 
+var bannedTermsHelpString = 'Please review viewport.js for a helper method or'
+  'mark with `/*OK*/` or `/*REVIEW*/` and consult the AMP team.';
+
+// Most of the forbidden property/method access banned on the
+// `forbiddenTermsSrcInclusive` object can be found in
+// [What forces layout / reflow gist by Paul Irish]
+// (https://gist.github.com/paulirish/5d52fb081b3570c81e3a)
+// These properties/methods when read/used require the browser
+// to have the up-to-date value to return which might possibly be an
+// expensive computation and could also be triggered multiple times
+// if we are not careful. Please mark the call with `object./*OK*/property`
+// if you explicitly need to read or update the forbidden property/method
+// or mark it `object./*REVIEW*/property` if you are unsure and so that it
+// stands out in code reviews.
+var forbiddenTermsSrcInclusive = {
+  '\\.innerHTML(?!_)': bannedTermsHelpString,
+  '\\.outerHTML(?!_)': bannedTermsHelpString,
+  '\\.postMessage(?!_)': bannedTermsHelpString,
+  '\\.scrollTop(?!_)': bannedTermsHelpString,
+  '\\.offsetLeft(?!_)': bannedTermsHelpString,
+  '\\.offsetTop(?!_)': bannedTermsHelpString,
+  '\\.offsetWidth(?!_)': bannedTermsHelpString,
+  '\\.offsetHeight(?!_)': bannedTermsHelpString,
+  '\\.offsetParent(?!_)': bannedTermsHelpString,
+  '\\.clientLeft(?!_)(?!_)': bannedTermsHelpString,
+  '\\.clientTop(?!_)': bannedTermsHelpString,
+  '\\.clientWidth(?!_)': bannedTermsHelpString,
+  '\\.clientHeight(?!_)': bannedTermsHelpString,
+  '\\.getClientRects(?!_)': bannedTermsHelpString,
+  '\\.getBoundingClientRect(?!_)': bannedTermsHelpString,
+  '\\.scrollBy(?!_)': bannedTermsHelpString,
+  '\\.scrollTo(?!_)': bannedTermsHelpString,
+  '\\.scrollIntoView(?!_)': bannedTermsHelpString,
+  '\\.scrollIntoViewIfNeeded(?!_)': bannedTermsHelpString,
+  '\\.scrollWidth(?!_)': 'please use `getScrollWidth()` from viewport',
+  '\\.scrollHeight(?!_)': bannedTermsHelpString,
+  '\\.scrollLeft(?!_)': bannedTermsHelpString,
+  '\\.scrollTop(?!_)': bannedTermsHelpString,
+  '\\.focus(?!_)': bannedTermsHelpString,
+  '\\.computedRole(?!_)': bannedTermsHelpString,
+  '\\.computedName(?!_)': bannedTermsHelpString,
+  '\\.innerText(?!_)': bannedTermsHelpString,
+  '\\.getComputedStyle(?!_)': bannedTermsHelpString,
+  '\\.scrollX(?!_)': bannedTermsHelpString,
```

Comment:
Since we are here, also need pageYOffset and pageXOffset - those are just synonyms of scrollY/X